warehouse/models/mart/payments/fct_elavon__transactions.sql

```diff
@@ -0,0 +1,98 @@
+{{ config(materialized='table') }}
+
+WITH stg_elavon__transactions AS (
+    SELECT * FROM {{ ref('stg_elavon__transactions') }}
+),
+
+remove_special_characters AS (
+    SELECT
+
+        * EXCEPT (fund_amt, batch_amt, amount, surchg_amount, convnce_amt, payment_date, transaction_date, settlement_date),
+
+        CAST(REGEXP_REPLACE(fund_amt, r'\$|,', '') as NUMERIC) AS fund_amt,
+        CAST(REGEXP_REPLACE(batch_amt, r'\$|,', '') as NUMERIC) AS batch_amt,
+        CAST(REGEXP_REPLACE(amount, r'\$|,', '') as NUMERIC) AS amount,
+        CAST(REGEXP_REPLACE(surchg_amount, r'\$|,', '') as NUMERIC) AS surchg_amount,
+        CAST(REGEXP_REPLACE(convnce_amt, r'\$|,', '') as NUMERIC) AS convnce_amt,
+
+        REGEXP_EXTRACT(payment_date, r'[^@\.]+') AS payment_date,
+        REGEXP_EXTRACT(transaction_date, r'[^@\.]+') AS transaction_date,
+        REGEXP_EXTRACT(settlement_date, r'[^@\.]+') AS settlement_date
+
+    FROM {{ ref('stg_elavon__transactions') }}
+),
+
+fct_elavon__transactions AS (
+    SELECT
+
+        * EXCEPT (payment_date, transaction_date, settlement_date),
+
+        CASE WHEN
+            LENGTH(payment_date) < 8
```

Comment:
Should this specifically check that the length is 7? Seems like this will only fix the formatting if the only thing missing is a leading 0 on the month. But if the length is 6 or less, don't think it will work (same comment for all these dates)